ehr_ml/clmbr/__init__.py

```diff
@@ -372,7 +373,80 @@ def test_helper(is_val: bool, sample_size: int) -> float:
                            os.path.join(model_dir, 'best'))
 
 
-def featurize_patients(model_dir: str, extract_dir: str, l: labeler.SavedLabeler) -> Tuple[np.array, np.array, np.array, np.array]:
+def featurize_patients(model_dir: str, extract_dir: str,
+                       patient_ids: Union[List[int], np.array],
+                       day_offsets: Union[List[int], np.array]) -> np.array:
+    """
+    Featurize patients using the given model.
+    This function will use the GPU if it is available.
+    """
+
+    config = read_config(os.path.join(model_dir, "config.json"))
+    info = read_info(os.path.join(model_dir, "info.json"))
+
+    use_cuda = torch.cuda.is_available()
+
+    model = PredictionModel(config, info, use_cuda=use_cuda).to(
+        device_from_config(use_cuda=use_cuda)
+    )
+    model_data = torch.load(os.path.join(model_dir, "best"), map_location="cpu")
+    model.load_state_dict(model_data)
+
+    final_transformation = partial(
+        PredictionModel.finalize_data,
+        config,
+        info,
+        device_from_config(use_cuda=use_cuda),
+    )
+
+    dummy_labels = [0 for _ in patient_ids]
+    data = (dummy_labels, patient_ids, day_offsets)
+
+    loaded_data = StrideDataset(
+        os.path.join(extract_dir, "extract.db"),
+        os.path.join(extract_dir, "ontology.db"),
+        os.path.join(model_dir, "info.json"),
+        data,
+        data
+    )
+
+    patient_id_to_info = defaultdict(dict)
+    for i, (pid, index) in enumerate(zip(patient_ids, day_offsets)):
+        patient_id_to_info[pid][index] = i
+
+    patient_representations = np.zeros((len(patient_ids), config['size']))
+    print(f'Total # patient days = {len(patient_ids)}', flush=True)
+    print(f'Total # batches = {(len(patient_ids)) / 10000}', flush=True)
+
+    # batch iterator args
+    is_val = True
+    batch_size = 10000
+    seed = info['seed']
+    threshold = config['num_first']
+    day_dropout = 0
+    code_dropout = 0
+    with dataset.BatchIterator(loaded_data, final_transformation,
+                               threshold=threshold, is_val=is_val,
+                               batch_size=batch_size, seed=seed,
+                               day_dropout=day_dropout,
+                               code_dropout=code_dropout) as batches:
+        pbar = tqdm(batches)
+        pbar.set_description('Computing patient representations')
+        for batch in pbar:
+             with torch.no_grad():
+                embeddings = (
+                    model.compute_embedding_batch(batch["rnn"]).cpu().numpy()
+                )
+                for i, patient_id in enumerate(batch["pid"]):
+                    for index, target_id in patient_id_to_info[patient_id].items():
+                        patient_representations[target_id, :] = embeddings[
+                            i, index, :
+                        ]
+
+    return patient_representations
+
+def featurize_patients_w_labels(model_dir: str, extract_dir: str,
```

Comment:
This should call featurize_patients to avoid duplicate code.

Reply:
Addressed in the latest commit.